workspace-manager-clienttests/src/main/java/scripts/uploadscripts/SummariesToBigQuery.java

```diff
@@ -90,13 +93,16 @@ public void uploadResults(
         // insert into testScriptResults
         tableId = TableId.of(datasetName, testScriptResultsTableName);
         InsertAllRequest.Builder insertRequestBuilder = InsertAllRequest.newBuilder(tableId);
-        for (int ctr = 0; ctr < renderedTestConfiguration.testScripts.size(); ctr++) {
-            TestScriptSpecification testScriptSpecification =
-                    renderedTestConfiguration.testScripts.get(ctr);
-            TestScriptResult.TestScriptResultSummary testScriptResult =
-                    testRunSummary.testScriptResultSummaries.get(ctr);
-            insertRequestBuilder.addRow(
-                    buildTestScriptResultsRow(testScriptSpecification, testScriptResult));
+        Map<String, TestScriptResult.TestScriptResultSummary> testScriptResultSummaries =
+                new ConcurrentHashMap<String, TestScriptResult.TestScriptResultSummary>();
+        testRunSummary.testScriptResultSummaries.stream()
+                .forEach(runSummary -> testScriptResultSummaries.put(runSummary.testScriptDescription, runSummary));
+
+        for (TestScriptSpecification testScriptSpecification : renderedTestConfiguration.testScripts) {
+            if (testScriptResultSummaries.containsKey(testScriptSpecification.name)) {
```

Comment:
as a future enhancement for TestRunner - if `TestScriptResultSummary.testScriptDescription` and `TestScriptSpecification.name` are intended to be equivalent, one of them should be renamed, i.e. use "name" for both classes or "description" for both classes, but be consistent.

Reply:
Fair enough, this is recorded in enhancement doc as well.